components/eam/src/physics/cam/zm_microphysics.F90

```diff
@@ -0,0 +1,3430 @@
+                                   +aero%mmrg_a(i,k,aero%coarse_no3_idx,aero%mode_coarse_dst_idx)) 
+
+#else
+                    dmc  = 0.5_r8*(aero%mmrg_a(i,k-1,aero%coarse_dust_idx,aero%mode_coarse_dst_idx)     &
+                                  +aero%mmrg_a(i,k,aero%coarse_dust_idx,aero%mode_coarse_dst_idx))
+                    ssmc = 0.5_r8*(aero%mmrg_a(i,k-1,aero%coarse_nacl_idx,aero%mode_coarse_slt_idx)     &
+                                  +aero%mmrg_a(i,k,aero%coarse_nacl_idx,aero%mode_coarse_slt_idx))  
+#endif
+                    if (aero%mode_coarse_idx > 0) then
+                       so4mc  = 0.5_r8*(aero%mmrg_a(i,k-1,aero%coarse_so4_idx,aero%mode_coarse_idx)     &
+                                   +aero%mmrg_a(i,k,aero%coarse_so4_idx,aero%mode_coarse_idx))
+                    endif
+
+#if (defined MODAL_AERO_4MODE_MOM || defined MODAL_AERO_5MODE)
+                    mommc  = 0.5_r8*(aero%mmrg_a(i,k-1,aero%coarse_mom_idx,aero%mode_coarse_idx)     &
+                                   +aero%mmrg_a(i,k,aero%coarse_mom_idx,aero%mode_coarse_idx))
+#endif
+
+#if (defined RAIN_EVAP_TO_COARSE_AERO)
+                    bcmc   = 0.5_r8*(aero%mmrg_a(i,k-1,aero%coarse_bc_idx,aero%mode_coarse_idx)     &
+                                   +aero%mmrg_a(i,k,aero%coarse_bc_idx,aero%mode_coarse_idx))
+                    pommc  = 0.5_r8*(aero%mmrg_a(i,k-1,aero%coarse_pom_idx,aero%mode_coarse_idx)     &
+                                   +aero%mmrg_a(i,k,aero%coarse_pom_idx,aero%mode_coarse_idx))
+                    soamc  = 0.5_r8*(aero%mmrg_a(i,k-1,aero%coarse_soa_idx,aero%mode_coarse_idx)     &
+                                   +aero%mmrg_a(i,k,aero%coarse_soa_idx,aero%mode_coarse_idx))
+#endif
+
+                    if (dmc > 0._r8) then
+                       if ( aero%separate_dust ) then
+                          ! 7-mode -- has separate dust and seasalt mode types and
+                          !           no need for weighting
+                          wght = 1._r8
+                       else
+                          ! 3-mode -- needs weighting for dust since dust and seasalt
+                          !           are combined in the "coarse" mode type
+#if (defined MODAL_AERO_4MODE_MOM && defined RAIN_EVAP_TO_COARSE_AERO )
+                          wght = dmc/(ssmc + dmc + so4mc + bcmc + pommc + soamc + mommc)
+#elif (defined MODAL_AERO_5MODE && defined RAIN_EVAP_TO_COARSE_AERO )
+                          wght = dmc/(ssmc + dmc + so4mc + bcmc + pommc + soamc + mommc)
+#elif (defined MODAL_AERO_4MODE_MOM)
+                          wght = dmc/(ssmc + dmc + so4mc + mommc)
+#elif (defined RAIN_EVAP_TO_COARSE_AERO)
+                          wght = dmc/(ssmc + dmc + so4mc + bcmc + pommc + soamc)
+#else
+                          wght = dmc/(ssmc + dmc) ! to keep FC5 binary identical
+#endif  
+                        endif
```

Comment:
@crterai @hwangacme not a big deal here. First, thanks for also adding the aerosol species for MOSAIC. Right now the calculation of wght does not have the condition compile flag for MOSAIC. We can do it later after ZMmicro is merged. Just want to make a note here.